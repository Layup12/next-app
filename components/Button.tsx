import { ButtonHTMLAttributes, FC } from "react";
import cn from "classnames";

import styles from "@styles/components/Button.module.scss";

interface IButton extends ButtonHTMLAttributes<HTMLButtonElement> {}

export const Button: FC<IButton> = ({ children, className, ...props }) => {
    return (
        <button className={cn(styles.root, className)} {...props}>
            {children}
        </button>
    );
};
