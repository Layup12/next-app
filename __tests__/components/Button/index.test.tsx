import { render, screen } from "@testing-library/react";

import { Button } from "@components/Button";

describe("Button", () => {
    it("snap initial button", () => {
        render(<Button>Initial</Button>);

        const button = screen.getByRole("button");

        expect(button).toMatchSnapshot();
    });
});
