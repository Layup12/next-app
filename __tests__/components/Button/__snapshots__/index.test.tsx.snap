// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snap initial button 1`] = `
<button
  class="root"
>
  Initial
</button>
`;
